runtime/platform.c

```diff
@@ -137,34 +138,28 @@ static uintnat round_up(uintnat size, uintnat align) {
   return (size + align - 1) & ~(align - 1);
 }
 
-long caml_sys_pagesize = 0;
+intnat caml_plat_pagesize = 0;
+intnat caml_plat_mmap_alignment = 0;
 
 uintnat caml_mem_round_up_pages(uintnat size)
 {
-  return round_up(size, caml_sys_pagesize);
+  return round_up(size, caml_plat_pagesize);
 }
 
-#ifdef _WIN32
-#define MAP_FAILED 0
-#endif
+#define Is_page_aligned(size) ((size & (caml_plat_pagesize - 1)) == 0)
 
 #ifdef DEBUG
 static struct lf_skiplist mmap_blocks = {NULL};
 #endif
 
 #ifndef _WIN32
-Caml_inline void safe_munmap(uintnat addr, uintnat size)
-{
-  if (size > 0)
-    munmap((void*)addr, size);
-}
 #endif
 
 void* caml_mem_map(uintnat size, uintnat alignment, int reserve_only)
 {
-  uintnat alloc_sz = caml_mem_round_up_pages(size + alignment);
-  void* mem;
-  uintnat base, aligned_start, aligned_end;
+  CAMLassert(Is_power_of_2(alignment));
+  CAMLassert(Is_page_aligned(size));
+  alignment = round_up(alignment, caml_plat_mmap_alignment);
```

Comment:
It is non-intuitive to me why we don't, instead, when `alignment < caml_plat_mmap_alignment`, set `alignment = 0`, which would give us non-trimming behavior when possible. I think that the reason is that we are not 100% sure that `mmap` returns `caml_plat_mmap_alignment`-aligned addresses? Wouldn't it be worth documenting this surprising non-guarantee somewhere so that people understand why we are doing things the complex way?